rsocket-core/src/main/java/io/rsocket/metadata/WellKnownMimeType.java

```diff
@@ -0,0 +1,128 @@
+  TEXT_PLAIN("text/plain", (byte) 33),
+  TEXT_XML("text/xml", (byte) 34),
+  VIDEO_H264("video/H264", (byte) 35),
+  VIDEO_H265("video/H265", (byte) 36),
+  VIDEO_VP8("video/VP8", (byte) 37),
+  // ... reserved for future use ...
+  MESSAGE_RSOCKET_TRACING_ZIPKIN("message/x.rsocket.tracing-zipkin.v0", (byte) 125),
+  MESSAGE_RSOCKET_ROUTING("message/x.rsocket.routing.v0", (byte) 126),
+  MESSAGE_RSOCKET_COMPOSITE_METADATA("message/x.rsocket.composite-metadata.v0", (byte) 127);
+
+  private final String str;
+  private final byte identifier;
+
+  WellKnownMimeType(String str, byte identifier) {
+    this.str = str;
+    this.identifier = identifier;
+  }
+
+  /** @return the byte identifier of the mime type, guaranteed to be positive or zero. */
+  public byte getIdentifier() {
+    return identifier;
+  }
+
+  /**
+   * @return the mime type represented as a {@link String}, which is made of US_ASCII compatible
+   *     characters only
+   */
+  public String getMime() {
+    return str;
+  }
+
+  /** @see #getMime() */
+  @Override
+  public String toString() {
+    return str;
+  }
+
+  /**
+   * Find the {@link WellKnownMimeType} for the given {@link String} representation. If the
+   * representation is {@code null} or doesn't match a {@link WellKnownMimeType}, the {@link
+   * #UNPARSEABLE_MIME_TYPE} is returned.
+   *
+   * @param mimeType the looked up mime type
+   * @return the matching {@link WellKnownMimeType}, or {@link #UNPARSEABLE_MIME_TYPE} if none
+   *     matches
+   */
+  public static WellKnownMimeType fromMimeType(String mimeType) {
+    if (mimeType == null) throw new IllegalArgumentException("type must be non-null");
+
+    // force UNPARSEABLE if by chance UNKNOWN_RESERVED_MIME_TYPE's text has been used
+    if (mimeType.equals(UNKNOWN_RESERVED_MIME_TYPE.str)) {
+      return UNPARSEABLE_MIME_TYPE;
+    }
+
+    for (WellKnownMimeType value : values()) {
+      if (mimeType.equals(value.str)) {
+        return value;
+      }
+    }
+    return UNPARSEABLE_MIME_TYPE;
+  }
+
+  /**
+   * Find the {@link WellKnownMimeType} for the given ID (as an int). Valid IDs are defined to be
+   * integers between 0 and 127, inclusive. IDs outside of this range will produce the {@link
+   * #UNPARSEABLE_MIME_TYPE}. Additionally, some IDs in that range are still only reserved and don't
+   * have a type associated yet: this method returns the {@link #UNKNOWN_RESERVED_MIME_TYPE} when
+   * passing such a ID, which lets call sites potentially detect this and keep the original
+   * representation when transmitting the associated metadata buffer.
+   *
+   * @param id the looked up id
+   * @return the {@link WellKnownMimeType}, or {@link #UNKNOWN_RESERVED_MIME_TYPE} if the id is out
+   *     of the specification's range, or {@link #UNKNOWN_RESERVED_MIME_TYPE} if the id is one that
+   *     is merely reserved but unknown to this implementation.
+   */
+  public static WellKnownMimeType fromId(int id) {
+    if (id < 0 || id > 127) {
```

Comment:
Checkout out how FrameType looks up FrameType via its code: 
https://github.com/rsocket/rsocket-java/blob/develop/rsocket-core/src/main/java/io/rsocket/frame/FrameType.java#L209

You can pre-compute the values into an array so you can look up using an array index instead of a for loop

Reply:
thanks for the pointer, done 👍

it looks like the same for the `String` based lookup could be improved with a static `Map<String, WellKnownMimeType>`:

```
Benchmark                                          Mode  Cnt         Score         Error  Units
WellKnownMimeTypePerf.fromStringMapLookup         thrpt   10  43906830.064 ± 1775994.988  ops/s
WellKnownMimeTypePerf.fromStringValuesLoopLookup  thrpt   10   2041048.116 ±   42704.445  ops/s
```
wdyt?

Reply:
yeah - that's a big difference